webClient/src/app/editor/code-editor/file-tabs/file-tabs.component.ts

```diff
@@ -38,17 +38,40 @@ export class FileTabsComponent implements OnInit {
     useBothWheelAxes: true
   };
 
-  constructor(@Inject(Angular2InjectionTokens.VIEWPORT_EVENTS) private viewportEvents: Angular2PluginViewportEvents) { }
+  private prevLength:number;
+
+  constructor(
+    private editorControl: EditorControlService,
+    @Inject(Angular2InjectionTokens.VIEWPORT_EVENTS) private viewportEvents: Angular2PluginViewportEvents) {}
 
   ngOnInit() {
     this.viewportEvents.resized.subscribe(()=> {
       this.componentRef.directiveRef.update();
     });
+    this.editorControl.initializedFile.subscribe(() => {
+      this.componentRef.directiveRef.scrollToRight();
+    });
+
+    console.log(this.componentRef.directiveRef.scrollToElement);
+    this.prevLength = 0;
+  }
+
+  ngAfterViewChecked() {
+    this.data.forEach((tab, i) => {
+      if (!tab.active) {
+        return;
+      }
+
+      this.componentRef.directiveRef.scrollToElement(`.tabs-file-list > li:nth-child(${i + 1})`);
```

Comment:
I have a new problem with this branch: I can't scroll my list (though my list can be auto-scrolled by closing & opening), and am having trouble clicking to change tabs for some tabs near the left or right sides.
I think it's this line of code, but not certain. Basically, you only would want to auto-scroll if there has been a change in the quantity, but here it's just after view check, which I believe could be for a different reason.